### PR TITLE
Refactor outdated README

### LINKED ISSUE #3076: Let's add reference to 7.x to README
Avoid confusion until 8.0 is out.

### PR DESCRIPTION
**Pull request by willingc**
Closes #3076

This PR provides better user guidance for ipywidgets for classic notebook and nbclassic.
- Surface where 7.6.3 release and docs exist
- Improve nav to master resources
- Add nav to 7.6.3
- Update grammar for clarity
- Rework compatibility table that is very outdated by making legacy everything before 7.6.3 and master

Backport to 7.x branch as well after merge.

### CODE REVIEW

**Comment by willingc** [comment]
Let's merge this as is since it provides more clarity than the existing outdated README. Folks can iterate on the compatibility table in a separate PR.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+| **Communication** | [![Join the chat at https://gitter.im/ipython/ipywidgets](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/jupyter-widgets/Lobby) [![Discourse](https://img.shields.io/badge/help_forum-discourse-blue?logo=discourse)](https://discourse.jupyter.org/) |
+| | |
 
-ipywidgets are [interactive HTML widgets](https://github.com/jupyter-widgets/ipywidgets/blob/master/docs/source/examples/Index.ipynb)
-for Jupyter notebooks, JupyterLab and the IPython kernel.
+**ipywidgets**, also known as jupyter-widgets or simply are [interactive HTML widgets](https://github.com/jupyter-widgets/ipywidgets/blob/master/docs/source/examples/Index.ipynb)
```
Comment (SylvainCorlay) [inline README.md:10]: Remove "or simply"?

File: README.md [hunk 2]
```diff
@@ -30,28 +33,27 @@ A [demonstration notebook](https://github.com/jupyter-widgets/ipywidgets/blob/ma
...
+following the current best practices for using interactive widgets. An
+implementation for a placeholder "Hello World" widget is provided as an example.
 
 Popular widget libraries such as
 [bqplot](https://github.com/bloomberg/bqplot),
 [pythreejs](https://github.com/jovyan/pythreejs) and
 [ipyleaflet](https://github.com/ellisonbg/ipyleaflet)
-
-follow exactly the same template and directory structure. They can serve as
+follow exactly the same template and directory structure. They serve as
```
Comment (SylvainCorlay) [inline README.md:53]: I just realize that bqplot now is in its own github organization (bqplot) while ipyleaflet and pythreejs are in the jupyter-widgets organization.

**Review by SylvainCorlay — APPROVED** [review]
Thanks! I think I found a typo and left another comment.

File: README.md (hunk 1 above)
Comment (willingc) [reply README.md:10]: Should be widgets. Good catch will update.

File: README.md (hunk 2 above)
Comment (willingc) [reply README.md:53]: I will update the links. Thanks.

**Pull request merged**